Userland/Libraries/LibChess/UCICommand.cpp

```diff
@@ -323,16 +324,181 @@ ErrorOr<String> BestMoveCommand::to_string() const
+            break;
+        } else if (name == "currline"sv) {
+            info_command->m_currline = TRY(parse_line(tokens.span().slice(i)));
+            break;
+        } else {
+            return Error::from_string_literal("Unknown info type");
+        }
+    }
+
+    return info_command;
 }
 
 ErrorOr<String> InfoCommand::to_string() const
 {
-    // FIXME: Implement this.
-    VERIFY_NOT_REACHED();
+    StringBuilder builder;
+
+    auto append_moves = [&](Vector<Chess::Move> const& moves) -> ErrorOr<void> {
+        bool first = true;
+        for (auto const& move : moves) {
+            if (!first)
+                TRY(builder.try_append(' '));
+
+            first = false;
+            TRY(builder.try_append(TRY(move.to_long_algebraic())));
+        }
+        return {};
+    };
+
+    TRY(builder.try_append("info"sv));
+    if (m_depth.has_value())
+        TRY(builder.try_appendff(" depth {}", m_depth.value()));
+    if (m_seldepth.has_value())
+        TRY(builder.try_appendff(" seldepth {}", m_seldepth.value()));
+    if (m_time.has_value())
+        TRY(builder.try_appendff(" time {}", m_time.value()));
+    if (m_nodes.has_value())
+        TRY(builder.try_appendff(" nodes {}", m_nodes.value()));
+    if (m_multipv.has_value())
+        TRY(builder.try_appendff(" multipv {}", m_multipv.value()));
+    if (m_score.has_value()) {
+        TRY(builder.try_append(" score"sv));
+        switch (m_score->type) {
+        case ScoreType::Centipawns:
+            TRY(builder.try_append(" cp"sv));
+            break;
+        case ScoreType::Mate:
+            TRY(builder.try_append(" mate"sv));
+            break;
+        }
+
+        TRY(builder.try_appendff(" {}", m_score->value));
+
+        switch (m_score->bound) {
+        case ScoreBound::None:
+            break;
+        case ScoreBound::Lower:
+            TRY(builder.try_append(" lowerbound"sv));
+            break;
+        case ScoreBound::Upper:
+            TRY(builder.try_append(" upperbound"sv));
+            break;
+        }
+    }
+    if (m_currmove.has_value())
+        TRY(builder.try_appendff(" currmove {}", TRY(m_currmove->to_long_algebraic())));
+    if (m_currmovenumber.has_value())
+        TRY(builder.try_appendff(" currmovenumber {}", m_currmovenumber.value()));
+    if (m_hashfull.has_value())
+        TRY(builder.try_appendff(" hashfull {}", m_hashfull.value()));
+    if (m_nps.has_value())
+        TRY(builder.try_appendff(" nps {}", m_nps.value()));
+    if (m_tbhits.has_value())
+        TRY(builder.try_appendff(" tbhits {}", m_tbhits.value()));
+    if (m_cpuload.has_value())
+        TRY(builder.try_appendff(" cpuload {}", m_cpuload.value()));
+    if (m_string.has_value())
+        TRY(builder.try_appendff(" string {}", m_string.value()));
+    if (m_pv.has_value()) {
+        TRY(builder.try_append(" pv "sv));
+        TRY(append_moves(m_pv.value()));
+    }
+
+    if (m_pv.has_value()) {
```

Comment:
This block is duplicated (almost, one has an extra space)

Reply:
No idea how that got there :see_no_evil: . Removed